### PR TITLE
Add representative count enrichment

### PR DESCRIPTION
**Pull request by lahsivjar**
Related to https://github.com/elastic/opentelemetry-dev/issues/378

### CODE REVIEW

File: enrichments/trace/internal/elastic/span.go [hunk 1]
```diff
@@ -391,6 +401,37 @@ func (s *spanEnrichmentContext) setDestinationService(span ptrace.Span) {
...
+		if pValue != "" {
+			p, _ = strconv.ParseUint(pValue, 10, 6)
+		}
+	}
+
+	if p == 63 {
+		// p-value == 63 represents zero adjusted count
+		return 0.0
+	}
+	return math.Pow(2, float64(p))
```
[the remark this answers is not in the document]
Comment (carsonip) [reply enrichments/trace/internal/elastic/span.go:432]: > it's not true as a p-value==64 will result in rep count of 0.

I assume you meant p==63 as stated in https://opentelemetry.io/docs/specs/otel/trace/tracestate-probability-sampling/#p-value

**Review by carsonip — APPROVED** [review]

File: enrichments/trace/internal/elastic/span.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lahsivjar) [reply enrichments/trace/internal/elastic/span.go:432]: > I assume you meant p==63 as stated in

Not quite. A p-value of `63` is a valid value, it represents a zero adjusted count (as per the specs you linked). OTOH, a p-value > 63 is invalid and should result in a rep count == `1` as per the comments in apm-data.

[the remark this answers is not in the document]
Comment (carsonip) [reply enrichments/trace/internal/elastic/span.go:432]: Got it. Do you consider that a bug in apm-data?

[the remark this answers is not in the document]
Comment (lahsivjar) [reply enrichments/trace/internal/elastic/span.go:432]: Honestly, the specs are a bit muddy around this so I don't think it would be a bug.

**Pull request merged**